File: cases/taint/java/expression-positive/ExpressionPositive.java
package dataflowbench.taint;

final class ExpressionPositive {
    static int dfb_source() { // DFB-SOURCE: expression-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: expression-sink

    static void run() {
        int value = dfb_source();
        int computed = (value * 3) + 7; // DFB-WITNESS: expression-computed
        dfb_sink(computed);
    }
}

File: cases/taint/kotlin/expression-positive/ExpressionPositive.kt
package dataflowbench

object ExpressionPositive {
    fun dfb_source(): Int { // DFB-SOURCE: expression-input
        return 1
    }

    fun dfb_sink(value: Int) {} // DFB-SINK: expression-sink

    fun run() {
        val value = dfb_source()
        val computed = (value * 3) + 7 // DFB-WITNESS: expression-computed
        dfb_sink(computed)
    }
}
